diff --git a/riscv-test-env/p/link.ld b/riscv-test-env/p/link.ld
index 187ef6f..2dd609d 100644
--- a/riscv-test-env/p/link.ld
+++ b/riscv-test-env/p/link.ld
@@ -3,16 +3,16 @@ ENTRY(_start)
 
 SECTIONS
 {
-  . = 0x80000000;
+  . = 0x00000000;
   .text.init : { *(.text.init) }
   . = ALIGN(0x1000);
-  .tohost : { *(.tohost) }
-  . = ALIGN(0x1000);
   .text : { *(.text) }
   . = ALIGN(0x1000);
   .data : { *(.data) }
   .data.string : { *(.data.string)}
   .bss : { *(.bss) }
+  . = 0xf0000100;
+  .tohost : { *(.tohost) }
   _end = .;
 }
 
